### PR TITLE
Add tests for `sf::Glsl` types

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

These tests are largely just the usual testing boilerplate but there are some meaningful tests towards the bottom with `Vec4`, `Mat3`, and `Mat4`.

### CODE REVIEW

**Review by binary1248 — APPROVED** [review]

**Comment by ChrisThrasher** [comment]
Might be worthwhile to mark `sf::priv::Vector4` as `constexpr` to match our 2D and 3D vectors.

**Comment by ChrisThrasher** [comment]
Code coverage shows 100% coverage of Glsl.inl and Glsl.cpp ✅

I'll merge this if CI ever succeeds.

**Comment by ChrisThrasher** [comment]
Hopefully https://github.com/SFML/SFML/pull/2932 fixes our codecov upload issues. This PR is on hold until we figure that out.

**Pull request merged**
